test/testother.cpp

```diff
@@ -5785,6 +5786,31 @@ class TestOther : public TestFixture {
                       errout.str());
     }
 
+    void duplicateExpression16() { //#10569
+        check("void f(const std::string& a) {\n"
+              "    if ((a == \"x\") ||\n"
+              "        (a == \"42\") ||\n"
+              "        (a == \"y\") ||\n"
+              "        (a == \"42\")) {}\n"
+              "}\n"
+              "void g(const std::string& a) {\n"
+              "    if ((a == \"42\") ||\n"
+              "        (a == \"x\") ||\n"
+              "        (a == \"42\") ||\n"
+              "        (a == \"y\")) {}\n"
+              "}\n"
+              "void h(const std::string& a) {\n"
+              "    if ((a == \"42\") ||\n"
+              "        (a == \"x\") ||\n"
+              "        (a == \"y\") ||\n"
+              "        (a == \"42\")) {}\n"
+              "}\n");
+        ASSERT_EQUALS("[test.cpp:1] -> [test.cpp:4]: (style) Same expression on both sides of '||'.\n"
```

Comment:
This message is not technically 100% true. It would be nice if we could rephraze the message somehow.

Duplicate expressions in condition?